zimui/src/stores/main.ts

```diff
@@ -1,52 +1,94 @@
-        },
-        (_) => {
+      try {
+        const response = await axios.get('./channel.json')
+        this.isLoading = false
+        this.channelData = response.data as Channel
+      } catch (error) {
+        if (error instanceof AxiosError) {
           this.isLoading = false
           this.channelData = null
-          this.errorMessage = 'Failed to load channel data'
-        },
-      )
+          this.handleAxiosError(error);
+          if (!axios.isAxiosError(error)) {
+            this.errorMessage = 'Failed to load channel data';
+          }
+          this.error = error // Set axios error to store
+        }
+      }
     },
     async fetchTopic(slug: string) {
       this.isLoading = true
       this.errorMessage = ''
-      return axios.get('./topics/' + slug + '.json').then(
-        (response) => {
-          this.isLoading = false
-          return response.data as Topic
-        },
-        (_) => {
+      try {
+        const response = await axios.get('./topics/' + slug + '.json')
+        this.isLoading = false
+        return response.data as Topic
+      } catch (error: unknown) {
+        if (error instanceof AxiosError) {
           this.isLoading = false
           this.channelData = null
-          this.errorMessage = 'Failed to load node ' + slug + ' data'
+          this.handleAxiosError(error);
+          if (!axios.isAxiosError(error)) {
+            this.errorMessage = 'Failed to load node ' + slug + ' data';
+          }
+          this.error = error // Set axios error to store
           return null
-        },
-      )
+        }
+      }
+    },
+    handleAxiosError(error: AxiosError<object>) {
+      this.isLoading = false;
+      this.channelData = null;
+      if (axios.isAxiosError(error)) {
+        if (error.response) {
+          const status = error.response.status;
+          switch (status) {
+            case 400:
+              this.errorMessage = 'Bad Request: The server could not understand the request due to invalid syntax.';
+              break;
+            case 401:
+              this.errorMessage = 'Unauthorized: Authentication is required and has failed or has not yet been provided.';
+              break;
+            case 403:
+              this.errorMessage = 'Forbidden: The server understood the request but refuses to authorize it.';
+              break;
+            case 404:
+              this.errorMessage = 'Not Found: The requested resource could not be found on the server.';
+              break;
+            case 500:
+              this.errorMessage = 'Internal Server Error: The server encountered an unexpected condition that prevented it from fulfilling the request.';
+              break;
+            default:
+              this.errorMessage = 'An error occurred: ' + error.message;
+              break;
+          }
+        } else {
+          this.errorMessage = 'Network Error: Unable to connect to the server.';
```

Comment:
Are we sure that "unable to connect to the server" is the only cause of `error.response` not being set? If not, we need a more generic message, a specific one like this would be misleaded

Reply:
I guess we should keep this one as it is for now, and when we encounter a particular case regarding this error, then we might want to modify this.

Reply:
No, this is not the way to do it. We are using this UI in a ZIM, the ZIM is then downloaded by someone to be used offline. The lead time between the moment when someone realize that the error message is inappropriate and the moment when he will get a new ZIM is very long (months or even sometimes years). We need to be as accurate as possible.

If you are unsure that when `error.response` is not set the only possible reason is that we failed to connect to the server, then please use a more generic error message reflecting the reality of what we know at this stage.